sdks/python/apache_beam/transforms/util.py

```diff
@@ -780,6 +783,48 @@ def expand(self, pcoll):
             self.max_buffering_duration_secs,
             self.clock))
 
+  @typehints.with_input_types(Tuple[K, V])
+  @typehints.with_output_types(Tuple[K, Iterable[V]])
+  class WithShardedKey(PTransform):
+    """A GroupIntoBatches transform that outputs batched elements associated
+    with sharded input keys.
+
+    The sharding is determined by the runner to balance the load during the
```

Comment:
How does this work? When/how runner does this load balancing?

Reply:
In Dataflow, the load balancing is done inside Windmill during the shuffle before the `GroupIntoBatchesDoFn`. Basically Windmill can re-distribute the input element to different workers and assign them different shard ids accordingly.

Different runner may choose different strategies by overriding the default implementation here. We will add the transform to the runner api (in follow-up PRs) so runners are able to recognize the transform and do something special if they want.

Reply:
For my understanding. So, windmill understands the coding for shardedkey and the implementation here is the default implementation for cases where runner does not do this overriding?

If this is correct, could you update the comment to explain this a bit more to the reader?

Reply:
Yes that is totally correct. I updated the documentation. How does it sound now?